code/iaas/allocator/src/main/java/io/cattle/platform/allocator/constraint/HostAffinityConstraint.java

```diff
@@ -33,16 +34,26 @@ public boolean matches(AllocationAttempt attempt,
         Set<Long> hostIds = candidate.getHosts();
         if (op == AffinityOps.SOFT_EQ || op == AffinityOps.EQ) {
             for (Long hostId : hostIds) {
-                Map<String, String> labelsForHost = allocatorDao.getLabelsForHost(hostId);
-                if (!labelValue.equals(labelsForHost.get(labelKey))) {
+                Map<String, String[]> labelsForHost = allocatorDao.getLabelsForHost(hostId);
```

Comment:
HostLabelMap object doesn't have the label's key/value.  just the host_id and label_id.